packages/driver/cypress/integration/cypress/log_spec.js

```diff
@@ -54,4 +54,51 @@ describe('src/cypress/log', function () {
       expect(result).to.equal(log)
     })
   })
+
+  context('countLogsByTests', () => {
+    it('returns zero if tests is empty', () => {
+      const tests = {}
+
+      expect(LogUtils.countLogsByTests(tests)).to.equal(0)
+    })
+
+    it('finds the highest id amongst the different types', () => {
+      const tests = {
+        a: {
+          agents: [{
+            id: 'log-idp.com-1',
+          }],
+          routes: [{
+            id: 'log-idp.com-2',
+          }],
+          commands: [{
+            id: 'log-idp.com-3',
+          }],
+          prevAttempts: [{
+            agents: [{
+              id: 'log-idp.com-4',
+            }],
+            routes: [{
+              id: 'log-idp.com-5',
+            }],
+            commands: [{
+              id: 'log-idp.com-6',
+            }],
+          }],
+        },
+      }
+
+      expect(LogUtils.countLogsByTests(tests)).to.equal(6)
+    })
+
+    it('returns zero if there are no agents routes or commands', () => {
```

Comment:
```suggestion
    it('returns zero if there are no agents, routes or commands', () => {
```

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/21137/commits/6a421f30f9b19dbef769d387ae224a018a01fea5